packages/host/app/resources/file.ts

```diff
@@ -106,8 +106,11 @@ class _FileResource extends Resource<Args> {
 
     if (newState.state === 'ready') {
       this.recentFilesService.addRecentFile(newState.url);
+      // code below handles redirect returned by the realm server
+      // this updates code path to be in-sync with the file.url
+      // For example, when inputting `drafts/author` will redirect to `drafts/author.gts`
       if (this._url != newState.url) {
-        this.operatorModeStateService.updateCodePath(new URL(newState.url));
+        this.operatorModeStateService.replaceCodePath(new URL(newState.url));
```

Comment:
Should file be coupled with operator mode service? What if you want to load a file outside the operator mode? Would it be more proper to do this in `onStateChange` callback in code mode component?

Reply:
yea this makes sense. I will pass this entire block in from the parent as an arg and remove the corresponding services. Uhmmmm I can see if `onStateChange` can work but there might be situation where the loading state gets in the way (it cud be a matter of changing the way I identify a redirect) -- lemme see

```
    if (newState.state === 'ready') {
      this.recentFilesService.addRecentFile(newState.url);
      // code below handles redirect returned by the realm server
      // this updates code path to be in-sync with the file.url
      // For example, when inputting `drafts/author` will redirect to `drafts/author.gts`
      if (this._url != newState.url) {
        this.operatorModeStateService.replaceCodePath(new URL(newState.url));
      }
    }
```

Reply:
I added an onRedirect hook into the file resource and removed all services. I think this is the clearest way. Changes of file resource I think are more generic than a redirect.